test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -587,3 +590,89 @@ def test_query_workspace_with_load_id_mass(setup_and_teardown):
             "includeMetadata": 1
         }
     )
+
+
+def test_query_workspace_with_load_id_mass_genome(setup_and_teardown):
+    # Test case 1: Valid scenario - test with genome objects and assembly_objs_only is set to False
+    ws = create_autospec(Workspace, spec_set=True, instance=True)
+    load_id = "998"
+    assembly_objs_response = {
+        "infos": [
+            [1, 'assembly_1', 'KBaseGenomeAnnotations.Assembly-6', 'time', 75, 'user', 42, 'wsname', 'md5', 78, {'foo': 'bar'}],
+            [2, 'assembly_2', 'KBaseGenomeAnnotations.Assembly-10', 'time', 75, 'user', 42, 'wsname', 'md5', 78, {}]
+        ]
+    }
+
+    genome_objs_response = {
+        "infos": [
+            [1, 'genome_1', 'KBaseGenomes.Genome-6', 'time', 75, 'user', 42, 'wsname', 'md5', 78,
+             {"load_id": load_id, "Assembly Object": "1/1/1"}],
+            [4, 'genome_2', 'KBaseGenomes.Genome-9.3', 'time', 75, 'user', 42, 'wsname', 'md5', 78,
+             {"load_id": load_id, "Assembly Object": "2/2"}]
+        ]
+    }
+
+    ws.get_object_info3.side_effect = [genome_objs_response, assembly_objs_response]
+
+    obj_names, obj_upas, assembly_objs_info, genome_objs_info = workspace_uploader._query_workspace_with_load_id_mass(
+        ws,
+        69046,
+        load_id,
+        ["genome_1", "genome_2",],
+        assembly_objs_only=False
+    )
+    assert obj_names == ["genome_1", "genome_2",]
+    assert obj_upas == ['42_1_75', '42_4_75']
+    assert assembly_objs_info == assembly_objs_response['infos']
+    assert genome_objs_info == genome_objs_response['infos']
+
+    # Assert expected calls to ws.get_object_info3
+    expected_calls = [
+        call({"objects": [{"wsid": 69046, "name": "genome_1"},
+                          {"wsid": 69046, "name": "genome_2"}], "ignoreErrors": 1, "includeMetadata": 1}),
+        call({"objects": [{"ref": "1/1/1"}, {"ref": "2/2"}], "includeMetadata": 1}),
+    ]
+    ws.get_object_info3.assert_has_calls(expected_calls, any_order=False)
```

Comment:
Does this check how many times ws.get_object_info3 get called?

Reply:
It doesn't, it just has to have the given calls. To ensure it only has those calls you can do 

`mock.call_count = 2`

Reply:
👍